math/README.md

```diff
@@ -39,6 +39,72 @@ Note that since `Shape2D` has `contains(Vector2)` method, `in` operator can be u
 (like `Rectangle`, `Ellipse` or `Circle`). For example, given `vec: Vector2` and `rect: Rectangle` variables, you can
 call `vec in rect` (or `vec !in rect`) to check if the rectangle contains (or doesn't) the point stored by the vector.
 
+#### `ImmutableVector2`
+- `ImmutableVector2` is an immutable equivalent to `Vector2`. It provides most of the functionality of `Vector2`, but
+mutation methods return new vectors instead of mutate the reference.
+- Note that one may want to create type aliases to makes the usage more concise: `typealias Vect2 = ImmutableVector2`
+- `ImmutableVector` is comparable (`>`, `>=`, `<`, `<=` are available). Comparison is evaluated by length
+- instances can be destructed: `val (x, y) = vector2`
+- `Vector2.toImmutableVector2()` Returns an immutable vector with same `x` and `y` attributes than this `Vector2`
+- `ImmutableVector2.toVector2()` Returns an mutable vector with same `x` and `y` attributes than this `ImmutableVector2`
+- Most of the functions of `Vector2` which mutate the vector are provided but deprecated. This allow smooth migration from
+`Vector2`. 
+- Notable difference with `Vector2`
+  - `+`, `-`, `*`, `/` are available and replace `add`, `sub` and `scl`.
```

Comment:
Mmmh.  But it is the case. Isn't it?

Reply:
Oh, I see. The README kind of suggests they are completely removed. Nevermind.

Reply:
Yes, the README kind of suggests that they are not available at all.... Which is somehow true, since deprecation level is "error", and the user cannot use them.

Reply:
Agreed, this is OK. I've written this comment before going thoroughly through the code, since the README shows up on the top of the page.